Added error prop

mohit-s96 opened the pull request:
closes #127 

- User of the Editor can now pass an `error` prop that will update the error state of the fixed editor.

mohit-s96 committed:
Added error prop

- User of the Editor can now pass an `error` prop that will update the error state of the fixed-editor.

AbhayVAshokan reviewed (changes requested):
@mohit-s96 _a please make the following changes and assign @karthiknmenon for the UI review.

https://vimeo.com/712110227/9aad5d97e8

mohit-s96 committed:
Added suggested changes
- Extracted error logic to HOC
- Added example for the error state
- Removed usage of tailwind classes and replaced them with scss classes.

karthiknmenon reviewed (changes requested):
@mohit-s96 _a Please check. 
Also, please match the error styles with that of neetoUI Input Component.

mohit-s96 committed:
Applied requested changes
